pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/tablestate/TableStateUtils.java

```diff
@@ -39,25 +39,25 @@ private TableStateUtils() {
   }
 
   /**
-   * Checks if all segments for the given @param tableNameWithType are succesfully loaded
-   * This function will get all segments in IDEALSTATE and CURRENTSTATE for the given table,
-   * and then check if all ONLINE segments in IDEALSTATE match with CURRENTSTATE.
-   * @param helixManager helix manager for the server instance
-   * @param tableNameWithType table name for which segment state is to be checked
-   * @return true if all segments for the given table are succesfully loaded. False otherwise
+   * Returns all online segments for a given table.
+   *
+   * @param helixManager instance of Helix manager
+   * @param tableNameWithType table for which we are obtaining ONLINE segments
+   *
+   * @return List of ONLINE segment names.
    */
-  public static boolean isAllSegmentsLoaded(HelixManager helixManager, String tableNameWithType) {
+  public static List<String> getOnlineSegmentsForThisInstance(HelixManager helixManager, String tableNameWithType) {
```

Comment:
Why not return this as a set?

Reply:
Because the code that used this earlier needed a list.

Reply:
On second thought, I think I could modify the original code to use a set as well, so reopening this. The only pitfall is that I start modifying code that has been working for some time, but seems simple enough t take the time to do this

Reply:
Addressed by commit e476c1efae63d35835576fbf0a75cf3b6c8ef970, @mcvsubbu please resolve the comment at your earliest